addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -66,10 +67,41 @@ public BodyGenerator(Generators generators) {
                         }
                     });
 
-    public String generate(String name, boolean isArray, List<String> refs) {
+    public String generateBodyWithObjectMaps(String name, boolean isArray, List<String> refs) {
         if (LOG.isDebugEnabled()) {
-            LOG.debug("Generate " + name);
+            LOG.debug("Generate body for object " + name);
         }
+        String jsonStr = generateJsonObjectString(name, refs);
+        if (isArray) {
+            jsonStr = convertJsonStringToArrayString(jsonStr);
+        }
+        return jsonStr;
+    }
+
+    public String generateBodyWithPrimitives(Property property, boolean isArray) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Generate body for primitive type " + property.getType());
+        }
+        String jsonStr = generateJsonPrimitiveValue(property);
+        if (isArray) {
+            jsonStr = convertJsonStringToArrayString(jsonStr);
+        }
+        return jsonStr;
+    }
+
+    private String convertJsonStringToArrayString(String jsonStr) {
```

Comment:
- This could be static (along with `SYNTAX`).
 - `createJsonArrayWith`? or something like that, seems to be more clear than "convert".